src/index.ts

```diff
@@ -36,10 +31,10 @@ export interface IDataAccessor<T> {
   (defaultValue: NonNullable<T>): NonNullable<T>;
 
   /**
-   * Data accessor with null default value.
-   * @param defaultValue
+   * Data accessor without a default value. If no data exists,
+   * `undefined` is returned.
    */
-  (nullDefaultValue: T extends null ? null : never): Defined<T>;
+  (): T | undefined
```

Comment:
This change breaks the test case mentioned above.

Reply:
Heh, this is why type tests are important too 😉